analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1290 @@
+    public void Method_Position(string invocation, int position, bool expected)
+    {
+        var snippet = $$"""
+            System.IFormatProvider provider = null;
+            {{invocation}}
+            """;
+        var context = ArgumentContextCS(WrapInMethodCS(snippet));
+
+        var descriptor = ArgumentDescriptor.MethodInvocation(KnownType.System_Int32, "ToString", "provider", position);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().Be(expected);
+    }
+
+    [DataTestMethod]
+    [DataRow("""i.ToString($$provider)""", 0, true)]
+    [DataRow("""i.ToString($$provider)""", 1, false)]
+    [DataRow("""i.ToString("", $$provider)""", 1, true)]
+    [DataRow("""i.ToString("", $$provider)""", 0, false)]
+    [DataRow("""i.ToString("", $$provider:= provider)""", 1, true)]
+    [DataRow("""i.ToString("", $$provider:= provider)""", 0, true)]
+    [DataRow("""i.ToString($$provider:= provider, format:= "")""", 1, true)]
+    [DataRow("""i.ToString($$provider:= provider, format:= "")""", 0, true)]
+    public void Method_Position_VB(string invocation, int position, bool expected)
+    {
+        var snippet = $$"""
+            Dim provider As System.IFormatProvider = Nothing
+            Dim i As Integer
+            {{invocation}}
+            """;
+        var context = ArgumentContextVB(WrapInMethodVB(snippet));
+
+        var descriptor = ArgumentDescriptor.MethodInvocation(KnownType.System_Int32, "ToString", "provider", position);
+        var result = MatchArgumentVB(context, descriptor);
+        result.Should().Be(expected);
+    }
+
+    [DataTestMethod]
+    [DataRow("""int.TryParse("", $$out var result);""")]
+    [DataRow("""int.TryParse("", System.Globalization.NumberStyles.HexNumber, null, $$out var result);""")]
+    public void Method_RefOut_True(string invocation)
+    {
+        var snippet = $$"""
+            System.IFormatProvider provider = null;
+            {{invocation}}
+            """;
+        var context = ArgumentContextCS(WrapInMethodCS(snippet));
+
+        var descriptor = ArgumentDescriptor.MethodInvocation(KnownType.System_Int32, "TryParse", "result", x => true, RefKind.Out);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().BeTrue();
+        context.Parameter.RefKind.Should().Be(RefKind.Out);
+    }
+
+    [DataTestMethod]
+    [DataRow("""Integer.TryParse("", $$result)""")]
+    [DataRow("""Integer.TryParse("", System.Globalization.NumberStyles.HexNumber, Nothing, $$result)""")]
+    public void Method_RefOut_True_VB(string invocation)
+    {
+        var snippet = $$"""
+            Dim provider As System.IFormatProvider = Nothing
+            Dim result As Integer
+            {{invocation}}
+            """;
+        var context = ArgumentContextVB(WrapInMethodVB(snippet));
+
+        var descriptor = ArgumentDescriptor.MethodInvocation(KnownType.System_Int32, "TryParse", "result", x => true, RefKind.Out);
```

Comment:
I wanted the API to force the developer to always specify constraints for the argument position because these are easy syntactic checks. If the argument position is unspecifiable, the developer should explicitly state that.